src/Build/BackEnd/Components/RequestBuilder/AssemblyLoadsTracker.cs

```diff
@@ -107,19 +83,47 @@ public void Dispose()
             StopTracking();
         }
 
+        private static bool IsBuiltinType(string? typeName)
+        {
+            if (string.IsNullOrEmpty(typeName))
+            {
+                return false;
+            }
+
+            return typeName!.StartsWith("Microsoft.Build", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
We should be using `StringComparison.Ordinal` here. Type names are case sensitive and known at compile time.

Reply:
This code can execute on top of the assembly filename as well (it's mounted prior actual loading of the task, when only the task name and some information about it's hosting assembly are known).
But since we are having consistent casing for our assembly files - it's good idea to be more strict here.